dace/sdfg/nodes.py

```diff
@@ -912,11 +912,10 @@ def used_symbols_within_scope(self, parent_state: 'dace.SDFGState', all_symbols:
                     )
                     free_symbols |= code_symbols
                     continue
+            elif isinstance(n, NestedSDFG):
```

Comment:
CFGs, Data Descriptors, and NestedSDFGs can have the function `used_symbols` defined. And since we check the nodes between map entry and exit, it needs to be a nested SDFG. This seems to be a correct fix for it.